### PR TITLE
[Util] Add collectSSRStyles util and update buildServerApp

### PR DESCRIPTION
**Pull request by damassi**
This adds a new utility `collectSSRStyles` which can be used to extract CSS styles from a styled component tree. Additionally, it updates `buildServerApp` to pass these styles back when we render an app. 

Additionally, it moves the [`<Container>`](https://github.com/artsy/force/blob/master/src/desktop/apps/artwork2/server.tsx#L33-L38) component located in force (and used in all new apps) over to Reaction. 

See contextual comments in the diff.

### CODE REVIEW

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit dfcc5fe89716a42ca81eaa77430ce17167b3972b

https://deploy-preview-1882--artsy-reaction.netlify.com

File: src/Apps/Artist/ArtistApp.tsx [hunk 1]
```diff
@@ -1,14 +1,15 @@
 import { Col, Row, Separator, Spacer } from "@artsy/palette"
 import { ArtistApp_artist } from "__generated__/ArtistApp_artist.graphql"
 import { NavigationTabsFragmentContainer as NavigationTabs } from "Apps/Artist/Components/NavigationTabs"
+import { AppContainer } from "Apps/Components/AppContainer"
+import { HorizontalPadding } from "Apps/Components/HorizontalPadding"
```
Comment (damassi) [inline src/Apps/Artist/ArtistApp.tsx]: These two components are used to frame new apps seen in the `/Apps` folder. In the future these can be combined into something like `<Layout>`

File: src/Apps/Artist/ArtistApp.tsx [hunk 2]
```diff
@@ -35,54 +36,56 @@ export class ArtistApp extends React.Component<ArtistAppProps> {
     const { artist, children } = this.props
 
     return (
-      <HorizontalPadding>
-        <Row>
-          <Col>
-            <ArtistHeader artist={artist} />
-          </Col>
-        </Row>
+      <AppContainer>
```
Comment (damassi) [inline src/Apps/Artist/ArtistApp.tsx]: Wraps the root component of our apps with a container that sets a max width. This used to be handled in force and was tech-debt that needed to be resolved.

File: src/Apps/Artwork/ArtworkApp.tsx [hunk 3]
```diff
@@ -105,80 +107,82 @@ export class ArtworkApp extends React.Component<Props> {
...
-        <Row>
-          <Col sm={8}>
-            <ArtworkBanner artwork={artwork} />
-            <Spacer mb={2} />
-          </Col>
-        </Row>
-
-        {/* Mobile */}
-        <Media at="xs">
+      <AppContainer>
```
Comment (damassi) [inline src/Apps/Artwork/ArtworkApp.tsx]: Same here (and all following `SomethingApp.tsx` files)

File: src/Apps/Components/AppContainer.tsx [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { breakpoints } from "@artsy/palette"
+import styled from "styled-components"
+
+export const AppContainer = styled.div`
+  width: 100%;
+  max-width: ${breakpoints.xl}px;
```
Comment (damassi) [inline src/Apps/Components/AppContainer.tsx]: Takes the largest possible breakpoint from our [palette Theme file](https://github.com/artsy/palette/blob/master/src/Theme.tsx#L18) and applies it here.

File: src/Artsy/Router/buildServerApp.tsx [hunk 5]
```diff
@@ -87,40 +81,36 @@ export function buildServerApp(config: ServerRouterConfig): Promise<Resolve> {
...
+            // Kick off relay requests to prime cache. TODO: Remove the need to
+            // do this by using persisted queries.
             ReactDOMServer.renderToString(<ServerApp />)
-            // Serializable data to be rehydrated on client
-            const data = await relayEnvironment.relaySSRMiddleware.getCache()
+            // Extract render queue for bundle split components using dyanamic `import()`
             const state = await getLoadableState(<ServerApp />)
-            return { relayData: data, loadableState: state }
+            // Extract CSS styleTags to inject for SSR pass
+            const tags = collectSSRStyles(<ServerApp />)
```
Comment (damassi) [inline src/Artsy/Router/buildServerApp.tsx]: This is basically the only change, aside from docs. It pulls out all of the CSS and returns a string of `<style>` tags that we can embed in the head of our page for server side rendering.

File: src/Artsy/Router/buildServerApp.tsx [hunk 6]
```diff
@@ -87,40 +81,36 @@ export function buildServerApp(config: ServerRouterConfig): Promise<Resolve> {
...
-            })
-          })
-        } catch (error) {
-          console.error(
-            "[Artsy/Router/buildServerApp] Error cleaning data",
-            error
-          )
-        }
+        // Strip response of problematic data structures
+        const relayData = cleanRelayData(_relayData)
```
Comment (damassi) [inline src/Artsy/Router/buildServerApp.tsx]: A little organizational cleanup, moves that block to a function

File: src/Artsy/Router/buildServerApp.tsx [hunk 7]
```diff
@@ -131,6 +121,7 @@ export function buildServerApp(config: ServerRouterConfig): Promise<Resolve> {
           ServerApp,
           status,
           headTags,
+          styleTags,
```
Comment (damassi) [inline src/Artsy/Router/buildServerApp.tsx]: Pass the new style tags back up, which is one can pluck out out [from here](https://github.com/artsy/force/blob/master/src/desktop/apps/artwork2/server.tsx#L21) when using the new router setup.

File: src/Artsy/Router/buildServerApp.tsx [hunk 8]
```diff
@@ -57,19 +58,12 @@ export function buildServerApp(config: ServerRouterConfig): Promise<Resolve> {
...
-            status,
-          })
+          resolve({ redirect })
           return
         }
 
         const headTags = [<style type="text/css">{MediaStyle}</style>]
-        const matchingMediaQueries =
-          userAgent && matchingMediaQueriesForUserAgent(userAgent)
+        const matchingMediaQueries = userAgent && matchingMediaQueriesForUserAgent(userAgent) // prettier-ignore
```
Comment (kierangillen) [inline src/Artsy/Router/buildServerApp.tsx]: Why was prettier causing an issue here?

Comment (damassi) [reply src/Artsy/Router/buildServerApp.tsx]: Sometimes when there's a bunch of variables in sequence and function names are long its slightly easier to read at a glance if things are uniform. Except for this purpose I rarely use `// prettier-ignore`

**Comment by kierangillen** [comment]
Looks good to me, do you want to resolve the conflicts then I'll merge?

**Pull request merged**